### ISSUE TITLE
Support for Alt-Text

### ISSUE THREAD

**Issue by db0** [asimons04/tesseract#7]
Lemmy supports adding alt-text to uploaded images. Can you add likewise for accessibility reasons?

**Comment by asimons04** [comment]
Yep.  It's on the roadmap.  I'm still targeting 0.19.3 as the minimum API (which doesn't have support for alt text), but I'm using the 0.19.7 JS client since it's backwards compatible. Recently I added the infrastructure to conditionally enable features based on the API level.

Do you recall which version of Lemmy API was the first to support alt text?  I want to say 0.19.4 but I could be mistaken.

**Comment by db0** [comment]
I think .5 but I might be mistaken as well :D

**Comment by asimons04** [comment]
I'll set the flag to .5 then to be safe.  I think .4 was kind of a stinker release that most people upgraded past pretty quickly. 

I *should* be able to work this into the upcoming 1.4.21 release.  It's been repeatedly delayed for various reasons, so what's a bit longer to add this in? lol

**Comment by asimons04** [comment]
Implemented in 2e3e03e4 to be included in the 1.4.21 release.

- Alt text field will dynamically appear in post form if the URL is to an image or direct video and the connected instance is on API version 0.19.5 or higher; hidden otherwise.
- Not sure how best to implement for other media since those do not have `alt` attributes.  Leaving those unimplemented / in the air for now.  May add some kind of paragraph text, but need to see what best practices are for those.
- If alt text is available from the API, will be used as alt text in post images; falls back to post title if `alt_text` not available (same as old behavior).
- Crossposts will use the alt text from the original post if available; can edit if desired/needed.
- Alt text is editable when editing posts.
